### PR TITLE
form page use active admin routes instead of resource_path helpers

### LINKED ISSUE #4544: "No route matches PATCH" or "DELETE" when using a model called "Collection"
I'm dealing with a model called `Collection` in ActiveAdmin. `index, new, show, edit` methods work absolutely fine, but `update` and `destroy` do not, because the `/:id` seems to be being omitted from the end of paths.

For instance, here is the opening code from the form tag when editing one of these models:

```
<form novalidate="novalidate" class="formtastic collection" id="edit_collection" action="/collections" accept-charset="UTF-8" method="post">
  <input name="utf8" type="hidden" value="✓">
  <input type="hidden" name="_method" value="patch">
```

If I use the inspector to alter the `action` attribute to `"/collections/3"`, the model updates successfully when the submit button is clicked.

(nb. I am not using a global namespace for ActiveAdmin here.)
### Workaround

Register the model under a different name with ActiveAdmin, eg.:

```
ActiveAdmin.register Collection as: 'collection_thing' do
  ...
```

### PR DESCRIPTION
**Pull request by Fivell**
form page use active admin routes instead of resource_path helpers

fixes #4544

### CODE REVIEW

**Comment by Fivell** [comment]
travis
```
undefined method `user' for #<Post:0x18575c1e> (ActionView::Template::Error)
```

connected with #4618 

Post ```has one author```  instead of ```user```, but it doesn't work this way

**Comment by Fivell** [comment]
@timoschilling , is that another bug?

**Comment by timoschilling** [comment]
I don't know

**Comment by Fivell** [comment]
@dallasgray, maybe you have idea why this happens ?

**Comment by Fivell** [comment]
@rdallasgray, maybe you have idea why this happens?

**Comment by rdallasgray** [comment]
@Fivell Nothing jumps out at me I'm afraid. Bear in mind @bamorim originated the code for nested `belongs_to` -- I just rebased and got tests passing.

File: lib/active_admin/resource/routes.rb [hunk 1]
```diff
@@ -98,13 +98,16 @@ def route_name(resource_path_name, options = {})
 
 
         # @return params to pass to instance path
-        def route_instance_params(instance)
-          belongs_to_names.reverse.reduce([instance]) do |arr,name| 
-            arr + [arr.last.public_send(name)]
+        def route_instance_params(instance, params = {})
+          belongs_to_names.reverse.reduce([instance.to_param]) do |arr, name|
+            #TODO read from belongs_to config
+            param_key = "#{name}_id"
```
Comment (rdallasgray) [inline lib/active_admin/resource/routes.rb]: Should this be a symbol as in line 111?

Comment (Fivell) [inline lib/active_admin/resource/routes.rb]: @rdallasgray , this is just another try I will rollback last changes

**Comment by Fivell** [comment]
@timoschilling , I think about continuing on this after we drop support rails 3.2, can't figure out how to make it work on all rails versions

**Comment by timoschilling** [comment]
ok, lets wait

**Comment by Fivell** [comment]
I will separate PRs after more deep investigation, I think it is better to create different PRs for different issues

**Comment by javierjulio** [comment]
@Fivell based on your last comment then, should we close this out?

**Comment by Fivell** [comment]
@javierjulio , this fix doesn't work for nested pages/resources and  requires belongs_to issues to be fixed first

**Pull request closed without merging**